src/main/java/org/radarcns/management/web/rest/DeviceTypeResource.java

```diff
@@ -52,6 +62,16 @@ public DeviceTypeResource(DeviceTypeService deviceTypeService) {
         if (deviceTypeDTO.getId() != null) {
             return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "idexists", "A new deviceType cannot already have an ID")).body(null);
         }
+        DeviceTypeDTO existing = deviceTypeService.findByProducerAndModel(
+            deviceTypeDTO.getDeviceProducer(), deviceTypeDTO.getDeviceModel());
+        if (existing != null) {
+            Map<String, String> errorParams = new HashMap<>();
+            errorParams.put("message", "A DeviceType with the specified producer and model "
+                + "already exists. This combination needs to be unique.");
+            errorParams.put("producer", deviceTypeDTO.getDeviceProducer());
+            errorParams.put("model", deviceTypeDTO.getDeviceModel());
```

Comment:
Wouldn't it be better if we send a `HttpResponse` than an exception? How would the client be noticed in case of the exception?

Reply:
The `ExceptionTranslator` takes care of transforming a raised exception to a proper HTTP response, with appropriate code. This is also the class that responds with `InternalServerError` by default for exceptions. By providing an implementation for a specific exception in `ExceptionTranslator` we can inform the client more clearly about why the request did not process successfully.